src/fe-common/core/fe-windows.c

```diff
@@ -189,26 +223,32 @@ void window_change_server(WINDOW_REC *window, void *server)
 
 void window_set_refnum(WINDOW_REC *window, int refnum)
 {
-	GSList *tmp;
+	GSequenceIter *oiter, *witer;
 	int old_refnum;
 
 	g_return_if_fail(window != NULL);
 	g_return_if_fail(refnum >= 1);
 	if (window->refnum == refnum) return;
 
-	for (tmp = windows; tmp != NULL; tmp = tmp->next) {
-		WINDOW_REC *rec = tmp->data;
+	oiter = g_sequence_lookup(windows_seq, GINT_TO_POINTER(refnum), (GCompareDataFunc)window_refnum_lookup, NULL);
```

Comment:
Please add some comments about what the `witer` and `oiter` are for, I guess they're the **W**indow iter and the **O**rigin (maybe **R**efnum **iter** is a clearer name).